level_tests.py

```diff
@@ -0,0 +1,33 @@
+import unittest
+from unittest.mock import patch, mock_open
+
+from level import Level
+
+class LevelTests(unittest.TestCase):
+
+    _text_moves_out_of_order = """
+level 1 5 5 3
+1:{0:[1]}
+0:{1:[2]}
+end_level
+"""
+
+    def setUp(self)
+        pass
+
+    @patch("builtins.open", new_callable=mock_open, read_data=_text_moves_out_of_order)
+    def test__get_next_event__moves_out_of_order__returns_correct(self, mock_level_file_open):
```

Comment:
For the return value, should this not be a parse error instead?

Reply:
I'd be happy to make this a parse error yes, in which case change the test. The parser could also accept events out-of-order but I agree that it's probably best it doesn't